### ISSUE TITLE
RF01 - issue when there's a join but no mixed columns

### ISSUE THREAD

**Issue by moseleyi** [sqlfluff/sqlfluff#6521]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

After last update I realised that a code like this:

```sql
SELECT
  test,
  table2.teste
FROM test
  LEFT JOIN test2 ON test.id = test2.id
```

Will fail, even though it might be correct. I understand the need for this rule to be enabled for referencing things that do not exist, like if I added `table3.test2` - a reference that doesn't exist. But in this case, even if it's best practice, it's not always necessary to prefix everything. I usually do that if there's a big mix of fields from both joins but if it's just one (like a dictionary table reference), then the rest of references I know are unique from the main table. 

Perhaps the rule could use a split into more concrete examples to allow us to configure this behaviour?

### Expected Behaviour

Ability to configure the rule to allow references without the table prefix

### Observed Behaviour

Rule fails

### How to reproduce

```sql
SELECT
  test,
  table2.teste
FROM test
  LEFT JOIN test2 ON test.id = test2.id
```

### Dialect

bigquery

### Version

3.3.0

### Configuration

[sqlfluff]
templater = dbt
dialect = bigquery
exclude_rules = RF02, AL01, ST07, ST06, LT14
; Default 80 chars is low considering lines that can contain window functions etc.
max_line_length = 150
large_file_skip_byte_limit = 30000

[sqlfluff:templater:dbt]
project_dir = ./

[sqlfluff:indentation]
tab_space_size = 2
indented_joins = true
indented_using_on = false

[sqlfluff:rules]
custom_rules = custom_rules::Rule_Element_LL99

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

[sqlfluff:rules:aliasing.length]
min_alias_length = 3

[sqlfluff:rules:references.keywords]
; Allow for calendar table
ignore_words = week, month, quarter, date

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by zagoodman** [comment]
I'm running into the same issue with a similar config (GBQ with dbt templater). I think it's a bug since the [docs for RF01](https://docs.sqlfluff.com/en/stable/reference/rules.html#rule-RF01) say:

> This rule is disabled by default for BigQuery, Databricks, Hive, Redshift, SOQL and SparkSQL due to the support of things like structs and lateral views which trigger false positives.

For now, I'm fixing by adding RF01 to `exclude_rules`.

**Comment by johalnes** [comment]
@zagoodman are there some issue directly related to your comment? Have RF01 issues now with databricks that suddenly appears on 3.3 and have seen a hugh chunk of code was removed regarding that exclusion by default

**Comment by peterbud** [comment]
RF01 is not applied anymore to such queries with the latest version
